my/coding/commits.py

```diff
@@ -112,18 +107,27 @@ def canonical_name(repo: Path) -> str:
     #     pass # TODO 
 
 
+def _fd_path() -> str:
+    fd_path: Optional[str] = shutil.which("fdfind") or shutil.which("fd-find") or shutil.which("fd")
+    if fd_path is None:
+        high(f"my.coding.commits requires 'fd' to be installed, See https://github.com/sharkdp/fd#installation")
+    # TODO: this just causes it to fail if 'fd' can't be found, but the warning is still sent... seems fine?
```

Comment:
I guess here it should be assert since it won't work otherwise anyway? (it would also make mypy happy without `or "fd"`)
In principle, there could be a fallback on `walk` or something later, similar to promnesia https://github.com/karlicoss/promnesia/blob/e44a2bda053cc93bb8dc1b84fd1a4461aa5d914f/src/promnesia/common.py#L437

Reply:
Yeah, I suppose an assert is a better error message than the long 'command not found'

Not sure how much overlap you have with helper methods like these in HPI and promnesia, not sure if it'd be too much work to split some parts of core into a separate package like `exporthelpers` or if that would just cause more complications.

Reply:
Yeah, not sure either -- with exporthelpers figured it was worth it since it was shared among so many (10+?) projects. And still, used submodules instead since maintaining & releasing extra package on pypi is kinda pain.
I guess two copies of code I can kind of tolerate for now -- maybe if it matures and later if HPI core is extracted it would make sense to just use it in Promnesia regardless whether any HPI modules are used.

Reply:
Yeah, thats about what I expected, makes sense